js/index.js

```diff
@@ -36,36 +22,71 @@ function status(text) {
 let websockifyUrl = new URL("../desktop-websockify/", window.location);
 websockifyUrl.protocol = window.location.protocol === "https:" ? "wss" : "ws";
 
-// Creating a new RFB object will start a new connection
-const rfb = new RFB(
-  document.getElementById("screen"),
-  websockifyUrl.toString(),
-  {},
-);
+/**
+ * Setup two way clipboard sync with the given rfb
+ *
+ * @param {RFB} rfb
+ */
+function setupClipboardSync(rfb) {
+  const clipboardText = document.getElementById("clipboard-text");
 
-// Add listeners to important events from the RFB module
-rfb.addEventListener("connect", connectedToServer);
-rfb.addEventListener("disconnect", disconnectedFromServer);
+  // Listen for clipboard events on the remote system, and automatically
+  // update our local textarea with those values
+  rfb.addEventListener("clipboard", (e) => {
+    clipboardText.value = e.detail.text;
+  });
 
-// Scale our viewport so the user doesn't have to scroll
-rfb.scaleViewport = true;
+  const clipboardClientChange = () => {
+    const text = clipboardText.value;
+    rfb.clipboardPasteFrom(text);
+  };
+  rfb.addEventListener("connect", () => {
+    console.log("connecting clipboard sync");
+    clipboardText.addEventListener("change", clipboardClientChange);
+  });
+  rfb.addEventListener("disconnect", () => {
+    console.log("disconnecting clipboard sync");
+    clipboardText.removeEventListener("change", clipboardClientChange);
+  });
+}
 
-// Use a CSS variable to set background color
-rfb.background = "var(--jupyter-medium-dark-grey)";
+function connect() {
+  // Creating a new RFB object will start a new connection
+  const rfb = new RFB(
+    document.getElementById("screen"),
+    websockifyUrl.toString(),
+    {},
+  );
 
-// Clipboard
-function clipboardReceive(e) {
-  document.getElementById("clipboard-text").value = e.detail.text;
-}
-rfb.addEventListener("clipboard", clipboardReceive);
+  // Update status when connection is made or broken
+  rfb.addEventListener("connect", () => {
+    console.log("connected");
+    setStatusText("Connected");
+  });
+  rfb.addEventListener("disconnect", () => {
+    console.log("disconnected");
+    let countDown = 5;
+    setStatusText(`Reconnecting in ${countDown}s`);
+    const intervalHandle = setInterval(() => {
+      countDown -= 1;
+      setStatusText(`Reconnecting in ${countDown}s`);
+      console.log(countDown);
+      if (countDown === 0) {
+        clearInterval(intervalHandle);
+        connect();
+      }
+    }, 1000);
```

Comment:
I think this could make sense to extract into a reconnect / attemptReconnect / handleDisconnect function or similar to improve readability.

Reply:
For reference, I saw a retry backoff example in https://www.codewithyou.com/blog/how-to-implement-retry-with-exponential-backoff-in-nodejs#implementing-retry-with-exponential-backoff-from-scratch